#338 Feat: 삭제 모달

JIN921 opened the pull request:
## 1. 무슨 이유로 코드를 변경했나요?
삭제 모달을 생성했습니다!

## 2. 어떤 위험이나 장애를 발견했나요?
![image](https://github.com/user-attachments/assets/cc047cef-a74b-4ffe-af9e-3a0d44f055bf)
메뉴 삭제 모달이 저런 위치에 뜨는 안타까운 일이 발생햇습니다...

메뉴 모달의 Content를   position: fixed; 으로 설정해 주어 일단 고쳤는데 이게 모바일에서는 잘 뜰지 모르겠네요,,,,ㅠㅠ 

마이페이지도 일단 고쳣어요
![image](https://github.com/user-attachments/assets/8a89ddbe-2bce-41a9-b09d-82784c96a69c)


## 3. 관련 스크린샷을 첨부해주세요.


## 4. 완료 사항
게시판 삭제 모달 
![image](https://github.com/user-attachments/assets/1c925c5d-b1b6-43e1-b3cf-2dfe604df7cd)

마이 페이지 탈퇴 모달
![image](https://github.com/user-attachments/assets/99f07aab-25dd-4410-ba43-eab9aefcd89e)

## 5. 추가 사항
- 탈퇴 모달 내용은 레퍼런스 보고,, 대충 적엇는데 이상하면 수정할게요!

yezzan9 commented:
> ![image](https://private-user-images.githubusercontent.com/84452202/413563615-cc047cef-a74b-4ffe-af9e-3a0d44f055bf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk2NDU3NTcsIm5iZiI6MTczOTY0NTQ1NywicGF0aCI6Ii84NDQ1MjIwMi80MTM1NjM2MTUtY2MwNDdjZWYtYTc0Yi00ZmZlLWFmOWUtM2EwZDQ0ZjA1NWJmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjE1VDE4NTA1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ3NmQ3OWE2YmJkNmE1ZDgxMWQwYmMxNGQ4NzI0MDcwNDI2NDlmMzE3ODJjZThhY2Q5ZDBmNTE4NGZkMGViNzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.XN6dnL7XNJcFKnwosOGr53G137VXyJ2z754W0IeaUDI) 메뉴 삭제 모달이 저런 위치에 뜨는 안타까운 일이 발생햇습니다...
> 
> 메뉴 모달의 Content를 position: fixed; 으로 설정해 주어 일단 고쳤는데 이게 모바일에서는 잘 뜰지 모르겠네요,,,,ㅠㅠ
> 
> 마이페이지도 일단 고쳣어요 ![image](https://private-user-images.githubusercontent.com/84452202/413564309-8a89ddbe-2bce-41a9-b09d-82784c96a69c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk2NDU3NTcsIm5iZiI6MTczOTY0NTQ1NywicGF0aCI6Ii84NDQ1MjIwMi80MTM1NjQzMDktOGE4OWRkYmUtMmJjZS00MWE5LWIwOWQtODI3ODRjOTZhNjljLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjE1VDE4NTA1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNiZTJmN2JlMGJiZDFkNDc3ZTk1Y2I0MWVlODM0ZDgyZDJiZWQ0MGVlYzBiNTZjZDY5YzgzMTg5NjRlYmY2N2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.jmsXrsZc9FabHVbvgSsiBl9aRStG1WbKVG0Q5IbV8QY)

이거 `position: fixed`로 설정했을 때 배경 부분 영역이 바뀌지 않나요?
따로 닫는 버튼 없이 배경에 onClose가 걸려있어서, 배경 위치가 변하지 않도록 `position: absolute`를 설정하고 top, right값을 props로 받아 위치를 변경하도록 구현해두었던거거든요!
배경에 색상 설정을 추가해서 확인해주실 수 있나요?

yezzan9 commented on src/components/Modal/DeleteModal.tsx:
```diff
@@ -0,0 +1,88 @@
+  title: string;
+  content: string;
+  buttonContent?: string;
+  onClose: () => void;
+  onDelete?: () => void;
+}) => {
```
onDelete가 필수값이 아닌 이유가 궁금합니다👀

JIN921 replied:
onDelete가 없어도 되는 모달이 될까봐.. 그랬는데 그냥 필수로 해야겠네요!

JIN921 commented:
> > ![image](https://private-user-images.githubusercontent.com/84452202/413563615-cc047cef-a74b-4ffe-af9e-3a0d44f055bf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk2NDU3NTcsIm5iZiI6MTczOTY0NTQ1NywicGF0aCI6Ii84NDQ1MjIwMi80MTM1NjM2MTUtY2MwNDdjZWYtYTc0Yi00ZmZlLWFmOWUtM2EwZDQ0ZjA1NWJmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjE1VDE4NTA1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ3NmQ3OWE2YmJkNmE1ZDgxMWQwYmMxNGQ4NzI0MDcwNDI2NDlmMzE3ODJjZThhY2Q5ZDBmNTE4NGZkMGViNzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.XN6dnL7XNJcFKnwosOGr53G137VXyJ2z754W0IeaUDI) 메뉴 삭제 모달이 저런 위치에 뜨는 안타까운 일이 발생햇습니다...
> > 메뉴 모달의 Content를 position: fixed; 으로 설정해 주어 일단 고쳤는데 이게 모바일에서는 잘 뜰지 모르겠네요,,,,ㅠㅠ
> > 마이페이지도 일단 고쳣어요 ![image](https://private-user-images.githubusercontent.com/84452202/413564309-8a89ddbe-2bce-41a9-b09d-82784c96a69c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk2NDU3NTcsIm5iZiI6MTczOTY0NTQ1NywicGF0aCI6Ii84NDQ1MjIwMi80MTM1NjQzMDktOGE4OWRkYmUtMmJjZS00MWE5LWIwOWQtODI3ODRjOTZhNjljLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjE1VDE4NTA1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNiZTJmN2JlMGJiZDFkNDc3ZTk1Y2I0MWVlODM0ZDgyZDJiZWQ0MGVlYzBiNTZjZDY5YzgzMTg5NjRlYmY2N2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.jmsXrsZc9FabHVbvgSsiBl9aRStG1WbKVG0Q5IbV8QY)
> 
> 이거 `position: fixed`로 설정했을 때 배경 부분 영역이 바뀌지 않나요? 따로 닫는 버튼 없이 배경에 onClose가 걸려있어서, 배경 위치가 변하지 않도록 `position: absolute`를 설정하고 top, right값을 props로 받아 위치를 변경하도록 구현해두었던거거든요! 배경에 색상 설정을 추가해서 확인해주실 수 있나요?

![image](https://github.com/user-attachments/assets/43b1deb5-48dd-4d1f-8e7a-0f7ddd86e390)
맞네용.. 배경이 밀리네요.. 근데 position: absolute 했을 때 위치가 이상해서.. 이거 뭐가 잘못된 거지.. 다시 확인해 보고 수정해 볼게요!

JIN921 commented:
> > ![image](https://private-user-images.githubusercontent.com/84452202/413563615-cc047cef-a74b-4ffe-af9e-3a0d44f055bf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk2NDU3NTcsIm5iZiI6MTczOTY0NTQ1NywicGF0aCI6Ii84NDQ1MjIwMi80MTM1NjM2MTUtY2MwNDdjZWYtYTc0Yi00ZmZlLWFmOWUtM2EwZDQ0ZjA1NWJmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjE1VDE4NTA1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ3NmQ3OWE2YmJkNmE1ZDgxMWQwYmMxNGQ4NzI0MDcwNDI2NDlmMzE3ODJjZThhY2Q5ZDBmNTE4NGZkMGViNzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.XN6dnL7XNJcFKnwosOGr53G137VXyJ2z754W0IeaUDI) 메뉴 삭제 모달이 저런 위치에 뜨는 안타까운 일이 발생햇습니다...
> > 메뉴 모달의 Content를 position: fixed; 으로 설정해 주어 일단 고쳤는데 이게 모바일에서는 잘 뜰지 모르겠네요,,,,ㅠㅠ
> > 마이페이지도 일단 고쳣어요 ![image](https://private-user-images.githubusercontent.com/84452202/413564309-8a89ddbe-2bce-41a9-b09d-82784c96a69c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk2NDU3NTcsIm5iZiI6MTczOTY0NTQ1NywicGF0aCI6Ii84NDQ1MjIwMi80MTM1NjQzMDktOGE4OWRkYmUtMmJjZS00MWE5LWIwOWQtODI3ODRjOTZhNjljLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjE1VDE4NTA1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNiZTJmN2JlMGJiZDFkNDc3ZTk1Y2I0MWVlODM0ZDgyZDJiZWQ0MGVlYzBiNTZjZDY5YzgzMTg5NjRlYmY2N2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.jmsXrsZc9FabHVbvgSsiBl9aRStG1WbKVG0Q5IbV8QY)
> 
> 이거 `position: fixed`로 설정했을 때 배경 부분 영역이 바뀌지 않나요? 따로 닫는 버튼 없이 배경에 onClose가 걸려있어서, 배경 위치가 변하지 않도록 `position: absolute`를 설정하고 top, right값을 props로 받아 위치를 변경하도록 구현해두었던거거든요! 배경에 색상 설정을 추가해서 확인해주실 수 있나요?

흠.. 오버레이 설정도 바꿔보고 하는데 오버레이가 화면에서 밀리지 않은 상태에서, 모달이 헤더 메뉴 버튼 아래에 위치 시키는 게 쉽지 않네요.. 전에는 어떻게,, 하신건가요,,?

yezzan9 commented:
> 흠.. 오버레이 설정도 바꿔보고 하는데 오버레이가 화면에서 밀리지 않은 상태에서, 모달이 헤더 메뉴 버튼 아래에 위치 시키는 게 쉽지 않네요.. 전에는 어떻게,, 하신건가요,,?

해당 컴포넌트가 top,right 값을 props로 받고 있는데, 컴포넌트를 사용하는 곳에서 그냥 1나1나 값을 설정해주었습니다..........
position: absolute인 상태에서요!

이렇게 하면 저희가 원하는 대로 보이긴 합니다..
모달의 위치를 조정하되 배경을 유지하는 게 너무 어렵더라구요🥵

JIN921 commented:
> > 흠.. 오버레이 설정도 바꿔보고 하는데 오버레이가 화면에서 밀리지 않은 상태에서, 모달이 헤더 메뉴 버튼 아래에 위치 시키는 게 쉽지 않네요.. 전에는 어떻게,, 하신건가요,,?
> 
> 해당 컴포넌트가 top,right 값을 props로 받고 있는데, 컴포넌트를 사용하는 곳에서 그냥 1나1나 값을 설정해주었습니다.......... position: absolute인 상태에서요!
> 
> 이렇게 하면 저희가 원하는 대로 보이긴 합니다.. 모달의 위치를 조정하되 배경을 유지하는 게 너무 어렵더라구요🥵

ㅠㅠ 근데... 이게 왜 저한테는 저렇게 밀려서 보일까요..? 설정해 두신 거 아무것도 건들지 않아도 저렇게 밀려보여서 수정을 한건데... 알 수 없다 css의 세계... 일단은 원래 설정대로 돌려 두겠읍니다..

yezzan9 reviewed (approved):
> ㅠㅠ 근데... 이게 왜 저한테는 저렇게 밀려서 보일까요..? 설정해 두신 거 아무것도 건들지 않아도 저렇게 밀려보여서 수정을 한건데... 알 수 없다 css의 세계... 일단은 원래 설정대로 돌려 두겠읍니다..

진짜 너무 슬퍼지네요....... 방법을 한번 고민해보겟습니다..
